solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrRecordWriter.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.common;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.format.StructuredRecordStringConverter;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrInputDocument;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * SolrRecordWriter - Instantiate a record writer that will build a Solr index.
+ */
+public class SolrRecordWriter extends RecordWriter<Text, Text> {
+  private static final String SERVER_URL = "solr.server.url";
+  private static final String SERVER_MODE = "solr.server.mode";
+  private static final String COLLECTION_NAME = "solr.server.collection";
+  private static final String KEY_FIELD = "solr.server.keyfield";
+  private static final String FIELD_MAPPINGS = "solr.output.field.mappings";
+  private static final String BATCH_SIZE = "solr.batch.size";
+  private static final Gson GSON = new Gson();
+  private final SolrSearchSinkConfig config;
+  List<SolrInputDocument> documentList = new ArrayList<SolrInputDocument>();
+  private SolrClient solrClient;
+  private Configuration conf;
+
+  public SolrRecordWriter(TaskAttemptContext context) {
+    conf = context.getConfiguration();
+    config = new SolrSearchSinkConfig(null, conf.get(SERVER_MODE), conf.get(SERVER_URL), conf.get(COLLECTION_NAME),
+                                      conf.get(KEY_FIELD), conf.get(FIELD_MAPPINGS));
+    solrClient = config.getSolrConnection();
+  }
+
+  @Override
+  public void write(Text key, Text value) throws IOException {
+    String solrFieldName;
+    SolrInputDocument document;
+    Type schemaType = new TypeToken<Schema>() { }.getType();
```

Comment:
put this object in a static, final field. No need to re-create this object for every record.